### ISSUE TITLE
Many tests timeout on FreeBSD

### ISSUE THREAD

**Issue by yurivict** [pika-org/pika#322]
Is this normal?

```
 244/1248 Test  #244: tests.unit.modules.algorithms.algorithms.reduce_by_key ....................................................***Timeout 1500.02 sec
          Start  245: tests.unit.modules.algorithms.algorithms.remove
 245/1248 Test  #245: tests.unit.modules.algorithms.algorithms.remove ...........................................................   Passed    0.04 sec
          Start  246: tests.unit.modules.algorithms.algorithms.remove1
 246/1248 Test  #246: tests.unit.modules.algorithms.algorithms.remove1 ..........................................................   Passed    0.04 sec
          Start  247: tests.unit.modules.algorithms.algorithms.remove2
 247/1248 Test  #247: tests.unit.modules.algorithms.algorithms.remove2 ..........................................................   Passed    0.05 sec
          Start  248: tests.unit.modules.algorithms.algorithms.remove_if
 248/1248 Test  #248: tests.unit.modules.algorithms.algorithms.remove_if ........................................................   Passed    0.06 sec
          Start  249: tests.unit.modules.algorithms.algorithms.remove_if1
 249/1248 Test  #249: tests.unit.modules.algorithms.algorithms.remove_if1 .......................................................***Timeout 1500.01 sec
          Start  250: tests.unit.modules.algorithms.algorithms.remove_copy
 250/1248 Test  #250: tests.unit.modules.algorithms.algorithms.remove_copy ......................................................***Timeout 1500.00 sec
          Start  251: tests.unit.modules.algorithms.algorithms.remove_copy_if
 251/1248 Test  #251: tests.unit.modules.algorithms.algorithms.remove_copy_if ...................................................   Passed    0.03 sec
          Start  252: tests.unit.modules.algorithms.algorithms.replace
 252/1248 Test  #252: tests.unit.modules.algorithms.algorithms.replace ..........................................................   Passed    0.03 sec
          Start  253: tests.unit.modules.algorithms.algorithms.replace_if
 253/1248 Test  #253: tests.unit.modules.algorithms.algorithms.replace_if .......................................................***Timeout 1500.01 sec
          Start  254: tests.unit.modules.algorithms.algorithms.replace_copy
 254/1248 Test  #254: tests.unit.modules.algorithms.algorithms.replace_copy .....................................................   Passed    0.03 sec
          Start  255: tests.unit.modules.algorithms.algorithms.replace_copy_if
 255/1248 Test  #255: tests.unit.modules.algorithms.algorithms.replace_copy_if ..................................................   Passed    0.04 sec
          Start  256: tests.unit.modules.algorithms.algorithms.reverse
 256/1248 Test  #256: tests.unit.modules.algorithms.algorithms.reverse ..........................................................***Timeout 1500.01 sec
          Start  257: tests.unit.modules.algorithms.algorithms.reverse_copy
 257/1248 Test  #257: tests.unit.modules.algorithms.algorithms.reverse_copy .....................................................   Passed    0.05 sec
          Start  258: tests.unit.modules.algorithms.algorithms.rotate
 258/1248 Test  #258: tests.unit.modules.algorithms.algorithms.rotate ...........................................................   Passed    0.06 sec
          Start  259: tests.unit.modules.algorithms.algorithms.rotate_copy
 259/1248 Test  #259: tests.unit.modules.algorithms.algorithms.rotate_copy ......................................................   Passed    0.05 sec
          Start  260: tests.unit.modules.algorithms.algorithms.search
 260/1248 Test  #260: tests.unit.modules.algorithms.algorithms.search ...........................................................***Timeout 1500.01 sec
          Start  261: tests.unit.modules.algorithms.algorithms.searchn
 261/1248 Test  #261: tests.unit.modules.algorithms.algorithms.searchn ..........................................................   Passed    0.04 sec
          Start  262: tests.unit.modules.algorithms.algorithms.set_difference
 262/1248 Test  #262: tests.unit.modules.algorithms.algorithms.set_difference ...................................................   Passed    0.10 sec
          Start  263: tests.unit.modules.algorithms.algorithms.set_intersection
 263/1248 Test  #263: tests.unit.modules.algorithms.algorithms.set_intersection .................................................   Passed    0.09 sec
          Start  264: tests.unit.modules.algorithms.algorithms.set_symmetric_difference
 264/1248 Test  #264: tests.unit.modules.algorithms.algorithms.set_symmetric_difference .........................................   Passed    0.09 sec
          Start  265: tests.unit.modules.algorithms.algorithms.set_union
 265/1248 Test  #265: tests.unit.modules.algorithms.algorithms.set_union ........................................................***Timeout 1500.01 sec
          Start  266: tests.unit.modules.algorithms.algorithms.shift_left
 266/1248 Test  #266: tests.unit.modules.algorithms.algorithms.shift_left .......................................................   Passed    0.06 sec
          Start  267: tests.unit.modules.algorithms.algorithms.shift_right
 267/1248 Test  #267: tests.unit.modules.algorithms.algorithms.shift_right ......................................................***Timeout 1500.01 sec
          Start  268: tests.unit.modules.algorithms.algorithms.sort
```

Overall it took 18+ hours to get to this point. Is this normal?

**Comment by msimberg** [comment]
@yurivict I take it from you running the tests that the patch in https://github.com/pika-org/pika/issues/314#issuecomment-1181476158 made things compile?

**Comment by yurivict** [comment]
> I take it from you running the tests that the patch in https://github.com/pika-org/pika/issues/314#issuecomment-1181476158 made things compile?

Yes.

**Comment by yurivict** [comment]
Now I observe that C++ compilation of something runs while unit tests are running.
Is this expected? This doesn't normally happen for most cmake projects.
I think timeouts are somehow related to this.
Maybe it sometimes rebuilds the whole project for unit tests?

**Comment by msimberg** [comment]
That is in fact expected, at least if you have `PIKA_WITH_TESTS_HEADERS` enabled. These tests check that headers include everything they need by themselves or transitively by compiling each header. Some tests are also of the kind "try to compile this file and check that it fails to compile". The first category generates a lot of tests. Especially if you're running tests in parallel that may end up using a lot of resources. The latter category is small.

**Comment by yurivict** [comment]
Tests don't time out in version 0.8.0

**Comment by msimberg** [comment]
@yurivict thanks for reporting back! I'll close this in that case for now. Feel free to reopen (or open a new issue) if you see similar failures in the future.
